graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -247,20 +143,20 @@ def _get_query_interval_of_ternary_integer_inequality_filter(parameter_values, f
             u"ternary integer inequality operator {}.".format(filter_operator)
         )
 
-    query_interval = _create_integer_interval(lower_bound, upper_bound)
-    return query_interval
+    return Interval[int](lower_bound, upper_bound)
 
 
-def _get_query_interval_of_integer_inequality_filter(parameter_values, filter_operator):
-    """Return IntegerInterval or None of values passing through a given integer inequality filter.
+def _get_query_interval_of_integer_inequality_filter(
+    parameter_values: List[int], filter_operator: str
+) -> Interval[int]:
+    """Return interval of values passing through a given integer inequality filter.
 
     Args:
-        parameter_values: List[int], describing the parameters for the inequality filter.
-        filter_operator: str, describing the inequality filter operation being performed.
+        parameter_values: the parameters for the inequality filter.
+        filter_operator: the inequality filter operation being performed.
 
     Returns:
-        - IntegerInterval namedtuple, non-empty interval of values that pass through the filter.
-        - None if the interval is empty.
+        interval of values that pass through the filter.
     """
```

Comment:
In terms of readability, it feels like this would be a better place to unpack the values from `parameter_values` — immediately inside each `len(parameter_values)` check. Then, the binary/ternary operator functions could take individual arguments for each parameter value, and we wouldn't need the asserts inside them since that part would be correct by construction.